### ISSUE TITLE
DeploymentNotFound with `daily`

### ISSUE THREAD

**Issue by weikanglim** [Azure/azure-dev#2119]
In one of our template-test runs, we saw one instance of `azd provision` failing with `DeploymentNotFound`. This is confusing and unexpected for the end user. This may be related to #2077 and Azure eventual consistency states being client-observable. Is there a defense in depth here we can apply?

https://dev.azure.com/azure-sdk/internal/_build/results?buildId=2756414&view=logs&j=014a604e-7c2f-5bb9-1866-e62d18d7fb76&t=6eb62864-2b51-5c47-1419-a8f03058a28f&l=4528

**Comment by weikanglim** [comment]
I logged this because conceptually I thought it was odd. We usually do run into *NotFound errors in other parts of the system, but running into this in provision is interesting.

Based on the error message "deploying to subscription", it seems like it's happening while polling for the deployment that was just submitted in [deployments.go](https://github.com/Azure/azure-dev/blob/5ba8fe5217835fa0ceccc924693b1cffdda65db7/cli/azd/pkg/tools/azcli/deployments.go#L164).  I'm also surprised that we just polled for a deployment we submitted, and the server said no.
